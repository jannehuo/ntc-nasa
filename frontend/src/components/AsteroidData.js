import * as React from 'react'
import { get } from 'lodash'
import formatNumber from '../helpers/numbers'
import '../scss/components/asterdoidData.scss'

function AsteroidData(props) {
  const { data } = props
  if (!data) {
    return null
  }
  const closeApproach = get(data, 'close_approach_data')
  return (
    <div className='asteroid-data-container'>
      <span className='sr-only'>Asteroid data ready</span>
      <div className='asteroid-card'>
        <div className='asteroid-card-header'>
          <i className='fas fa-meteor' aria-hidden='true'></i>
          <div className='asteroid-card-header-size'>
            Estimated:{' '}
            {formatNumber(
              get(
                data,
                'estimated_diameter.kilometers.estimated_diameter_max',
                ''
              )
            )}{' '}
            km
          </div>
        </div>
        <div className='asteroid-card-data'>
          <div className='asteroid-card-data-info'>
            <h2>{get(data, 'name')}</h2>
            <ul className='asteroid-card-data-info-list'>
              <li>
                <p className='asteroid-card-data-info-list-label'>Kilometers</p>
                <p className='asteroid-card-data-info-list-value'>
                  {formatNumber(
                    get(
                      data,
                      'estimated_diameter.kilometers.estimated_diameter_max',
                      ''
                    )
                  )}
                </p>
              </li>
              <li>
                <p className='asteroid-card-data-info-list-label'>Meters</p>
                <p className='asteroid-card-data-info-list-value'>
                  {formatNumber(
                    get(
                      data,
                      'estimated_diameter.meters.estimated_diameter_max',
                      ''
                    )
                  )}
                </p>
              </li>
              <li>
                <p className='asteroid-card-data-info-list-label'>Magnitude</p>
                <p className='asteroid-card-data-info-list-value'>
                  {formatNumber(get(data, 'absolute_magnitude_h', ''))}
                </p>
              </li>
            </ul>
            {closeApproach &&
              closeApproach.map((approachData, i) => (
                <ul
                  className='asteroid-card-data-info-approach-info-list'
                  key={i}
                >
                  <li>
                    <p>Date</p>
                    <p>{get(approachData, 'close_approach_date', '')}</p>
                  </li>
                  <li>
                    <p>Speed</p>
                    <p>
                      {formatNumber(
                        get(
                          approachData,
                          'relative_velocity.kilometers_per_hour'
                        )
                      )}{' '}
                      km/h
                    </p>
                  </li>
                  <li>
                    <p>Distance</p>
                    <p>
                      {formatNumber(
                        get(approachData, 'miss_distance.kilometers', '')
                      )}{' '}
                      km
                    </p>
                  </li>
                </ul>
              ))}
          </div>
        </div>
      </div>
    </div>
  )
}

export default AsteroidData
