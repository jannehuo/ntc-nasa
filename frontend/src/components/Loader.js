import React from 'react'
import '../scss/components/loader.scss'

function Loader(props) {
  const { visible } = props
  if (visible) {
    return (
      <div className='loader-container'>
        <h1>Fetching data. Please wait.</h1>
        <div className='loader-icon-container'>
          <i className='fas fa-meteor' aria-hidden='true'></i>
        </div>
      </div>
    )
  }
  return null
}

export default Loader
