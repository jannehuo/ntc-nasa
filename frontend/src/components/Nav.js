import * as React from 'react'
import { FRONTPAGE, YEAR } from '../paths'
import { Link } from 'react-router-dom'
import '../scss/components/nav.scss'

function Nav() {
  return (
    <div className='nav-container'>
      <h1 className='project-header'>
        <Link to={`${FRONTPAGE}`}>
          <i className='fas fa-globe' aria-hidden='true'></i>
          <span>NTC Nasa</span>
        </Link>
      </h1>
      <nav>
        <ul className='nav-links'>
          <li>
            <Link to={`${FRONTPAGE}`}>Frontpage</Link>
          </li>
          <li>
            <Link to={`${YEAR}`}>Find by year</Link>
          </li>
        </ul>
      </nav>
    </div>
  )
}

export default Nav
