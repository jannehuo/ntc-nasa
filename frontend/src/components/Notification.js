import React from 'react'
import '../scss/components/notification.scss'

const messages = {
  503: 'Backend service is unavailable',
  429: 'Too Many Requests',
}

function Notification(props) {
  const { data } = props
  const { visible, status } = data
  const defaultMessage = 'Something went wrong. Try again!'
  const message = messages[status] || defaultMessage
  if (visible) {
    return (
      <div className='notification-container'>
        <i className='fas fa-exclamation-circle'></i>
        <span>{message}</span>
      </div>
    )
  }

  return null
}

export default Notification
