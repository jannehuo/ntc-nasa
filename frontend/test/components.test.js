import React from 'react'
import renderer from 'react-test-renderer'
import { HashRouter as Router } from 'react-router-dom'
import asteroidData from './mockData/astedoid'
import AsteroidData from '../src/components/AsteroidData'
import Loader from '../src/components/Loader'
import Notification from '../src/components/Notification'
import Nav from '../src/components/Nav'

describe('Test frontpage', () => {
  it('Should render asteroid data', () => {
    const component = renderer.create(<AsteroidData data={asteroidData} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Should render loader', () => {
    const component = renderer.create(<Loader visible={true} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Should render notification for status 503', () => {
    const data = {
      visible: true,
      status: 503,
    }
    const component = renderer.create(<Notification data={data} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Should render notification for status 429', () => {
    const data = {
      visible: true,
      status: 429,
    }
    const component = renderer.create(<Notification data={data} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Should render notification for status 500', () => {
    const data = {
      visible: true,
      status: 500,
    }
    const component = renderer.create(<Notification data={data} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Should render navigation', () => {
    const component = renderer.create(
      <Router>
        <Nav />
      </Router>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
