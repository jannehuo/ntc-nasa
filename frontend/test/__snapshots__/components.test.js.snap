// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test frontpage Should render asteroid data 1`] = `
<div
  className="asteroid-data-container"
>
  <span
    className="sr-only"
  >
    Asteroid data ready
  </span>
  <div
    className="asteroid-card"
  >
    <div
      className="asteroid-card-header"
    >
      <i
        aria-hidden="true"
        className="fas fa-meteor"
      />
      <div
        className="asteroid-card-header-size"
      >
        Estimated:
         
        1.58
         
        km
      </div>
    </div>
    <div
      className="asteroid-card-data"
    >
      <div
        className="asteroid-card-data-info"
      >
        <h2>
          22099 (2000 EX106)
        </h2>
        <ul
          className="asteroid-card-data-info-list"
        >
          <li>
            <p
              className="asteroid-card-data-info-list-label"
            >
              Kilometers
            </p>
            <p
              className="asteroid-card-data-info-list-value"
            >
              1.58
            </p>
          </li>
          <li>
            <p
              className="asteroid-card-data-info-list-label"
            >
              Meters
            </p>
            <p
              className="asteroid-card-data-info-list-value"
            >
              1577.76
            </p>
          </li>
          <li>
            <p
              className="asteroid-card-data-info-list-label"
            >
              Magnitude
            </p>
            <p
              className="asteroid-card-data-info-list-value"
            >
              17.88
            </p>
          </li>
        </ul>
        <ul
          className="asteroid-card-data-info-approach-info-list"
        >
          <li>
            <p>
              Date
            </p>
            <p>
              2015-12-20
            </p>
          </li>
          <li>
            <p>
              Speed
            </p>
            <p>
              27010.82
               
              km/h
            </p>
          </li>
          <li>
            <p>
              Distance
            </p>
            <p>
              46415376.53
               
              km
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test frontpage Should render loader 1`] = `
<div
  className="loader-container"
>
  <h1>
    Fetching data. Please wait.
  </h1>
  <div
    className="loader-icon-container"
  >
    <i
      aria-hidden="true"
      className="fas fa-meteor"
    />
  </div>
</div>
`;

exports[`Test frontpage Should render navigation 1`] = `
<div
  className="nav-container"
>
  <h1
    className="project-header"
  >
    <a
      href="#/"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fas fa-globe"
      />
      <span>
        NTC Nasa
      </span>
    </a>
  </h1>
  <nav>
    <ul
      className="nav-links"
    >
      <li>
        <a
          href="#/"
          onClick={[Function]}
        >
          Frontpage
        </a>
      </li>
      <li>
        <a
          href="#/year"
          onClick={[Function]}
        >
          Find by year
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Test frontpage Should render notification for status 429 1`] = `
<div
  className="notification-container"
>
  <i
    className="fas fa-exclamation-circle"
  />
  <span>
    Too Many Requests
  </span>
</div>
`;

exports[`Test frontpage Should render notification for status 500 1`] = `
<div
  className="notification-container"
>
  <i
    className="fas fa-exclamation-circle"
  />
  <span>
    Something went wrong. Try again!
  </span>
</div>
`;

exports[`Test frontpage Should render notification for status 503 1`] = `
<div
  className="notification-container"
>
  <i
    className="fas fa-exclamation-circle"
  />
  <span>
    Backend service is unavailable
  </span>
</div>
`;
